tests/ui/lint/non_local_definitions.stderr

```diff
@@ -19,7 +19,7 @@ warning: non-local `impl` definition, they should be avoided as they go against
 LL |     impl Uto for *mut Test {}
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^
    |
-   = help: move this `impl` block outside the of the current constant expression `<unnameable>`
+   = help: move this `impl` block outside the of the current type alias `A`
```

Comment:
Hum... This appears to be a correctness issue.

It appears that HIR Owner != `tcx.parent`.

This example should lint, but doesn't with HIR owner since with `parent_owner_iter` the parent of the `impl` def is the anon-const instead of the closure with `tcx.parent`.

```rust
const _: u32 = {
    let _a = || {
        impl Uto6 for Test {}
        //~^ WARN non-local `impl` definition

        1
    };
}
```